S3TransferUtilitySample/src/com/amazonaws/demo/s3transferutility/UploadActivity.java

```diff
@@ -519,85 +517,68 @@ private void beginUploadInBackground(String filePath) {
     }
 
     /*
-     * Gets the file path of the given Uri.
+     * Copies the resource associated with the Uri to a new File in the cache directory.
+     * @param uri The Uri.
+     * @return a new File representing the Uri in the cache directory.
+
      */
```

Comment:
So, are you writing a new file in the cache folder each time an `upload` is attempted? Ugh, it seems that [#1436](https://github.com/aws-amplify/aws-sdk-android/issues/1436) is more relevant than ever now in API 29+

Reply:
Oh yeah, I hadn't even see [#1436](https://github.com/aws-amplify/aws-sdk-android/issues/1436), but I had the same thought.  It is something that should be addressed for sure.